Service Borger Upri- Kitchen Upgrades

Dawnseer opened the pull request:
## About The Pull Request
I decided that Service Borgs had been left in the dust for too long, and that since they're named 'Service' cyborgs and not 'Bar' cyborgs I'd give them upgrades to let them assist in the kitchen. To do this I've made 4 upgrades. 

First up, rolling table dock, to provide mobile service. Nothing worse than offering someone a drink in a hallway and having to drag them to the nearest table or having to drop it on the floor. That's poor service. 
Secondly there's a condiment synthesizer, which can produce solid condiments like flour, cornmeal, salt, coco powder, etc. 
  Thirdly a kitchen toolset, a knife that can switch to a rolling pin. The blade dulls when not used for cooking.
Fourthly, A service apparatus, which can pick up food, oven trays, soup pots, plates, bowls and paper.

The upgrades are locked behind a new techweb node, Cyborg Upgrades: Service.
This node is locked behind the same nodes as the janitorial cyborg upgrade node.

And outside the upgrades, I've also given the cyborg a rag so it can finally clean it's glasses and tables. I've also given it a money bag, to store the profits. This gives cyborgs an opportunity to save up for getting on the deluxe shuttle. Or giving people a new target to hold up and shake down for their hard earned money. Please don't law 2 them into giving you their lunch money. Or do.
I've also enabled the RSF  to print a few more things. Plates, bowls, plastic cutlery and paper cups.
I've also enabled borgs to open and close the oven. Alas, they still can't pick up food or the oven trays from the oven, since the oven is sticky in a manner.

I've also added a little bit more to the borgshaker, and moved sugar from the borgshaker to the condiment synthesizer.

And not only the borgshaker, but the emagged shaker too. The emagged shaker has been given blood and carpotoxin. And you may ask 'Why those two ingredients?' Both can be used like fernet, both for nice things, and harmful things, staying in line with fernet. Blood can be used to mix drinks or be used for food, or for  disease, since the borg or anyone with access to milk and water can easily produce virus food. Meanwhile carpotoxin is well, a rather weak toxin but can also be used to make imitation carpmeat. This'l give more reason to also consider the borg for emagging, or giving illegal tech, instead of just speedlining for the nearest engineering borg or medical borg.
![image](https://user-images.githubusercontent.com/126404225/233840472-7665196c-1052-4e29-9749-6c2b99e5fa9e.png)

The Condiment Synthesizer uses the same UI as the medical hypo, due to it's ingredient list being low in comparison to the shaker
![image](https://user-images.githubusercontent.com/126404225/235359571-48157d24-f3c8-448f-bb05-6f8842526aae.png)

I've been trying to figure out how to get an item to allow the borg to open the cooking menu when used inhand/inmodule. But I can't figure that one out, sadly. Nor how to let the Service Apparatus pick up trays from the oven.
So for now, service borgs will have to settle for simply assisting in the kitchen.

## Why It's Good For The Game
Allows Service Cyborgs the tools to help in the kitchen, and elsewhere too. 
## Changelog
:cl:
add: Damp rag and Money bag to service borg
add: Service borg upgrades, with accompanying tech web node
add: Sprites for kitchen toolset, and service apparatus
balance: The RSF can print a few more things, paper cups, seaweed sheets, plates, bowls, plastic cutlery and standard decks of cards. 
balance: Sugar moved from borgshaker to condiment synthesizer. Berry juice, cherry jelly and vinegar added to borgshaker. balance: Both shaker and synthesizer can also be adjusted to give 1 unit at time, so it goes 5, 10, 20 then 1, and back around.
balance: Adds blood and Carpotoxin to the emagged borgshaker.
fix: Money bag was missing a description, it now has one.
/:cl:

Dawnseer committed:
Borger uprising

Yeah okay next time I am going to make a commit for every change instead of making one big change this is hard to organize

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
empty space n fix

Dawnseer committed:
Merge branch 'ServiceBorgUpgrades' of https://github.com/Dawnseer/tgstation into ServiceBorgUpgrades

Dawnseer committed:
Whoops

Jezzared commented:
Borg rag! Long live the Borg rag!

Finally, i can clean the glasses after the meatbags contaminate them.

Dawnseer committed:
Come on now

AtomTheProphet commented:
Wooo! I love it! This is exactly what it needs.

datorangebottle commented:
Are they able to pick stuff up off of / put stuff down on the griddles/ovens?
For some reason I remember this not being a thing in the past.

Dawnseer commented:
> Are they able to pick stuff up off of / put stuff down on the griddles/ovens? For some reason I remember this not being a thing in the past.

No, I made an apparatus that could pick up oven trays and put them in the oven, but getting the oven to release the tray from it's cruel grip was beyond me. So for now the apparatus is hidden in my project folder, same as my attempt at making an item open up the cooking menu.

GoldenAlpharex commented:
Just letting you know, for the knife/rolling pin stuff, downstream we just made a "power tool" cyborg version of that, which would probably be a better alternative here as it's just two slots instead of one. It also had sprites for both endings, now whether or not you'd like to keep that one or not would be entirely up to you.

If you're interested, take a look at the [code for it](https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/modules/borg_buffs/code/robot.dm#L114). The sprites are [here](https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/modules/borg_buffs/icons/items_cyborg.dmi).

Dawnseer commented:
> Just letting you know, for the knife/rolling pin stuff, downstream we just made a "power tool" cyborg version of that, which would probably be a better alternative here as it's just two slots instead of one. It also had sprites for both endings, now whether or not you'd like to keep that one or not would be entirely up to you.
> 
> If you're interested, take a look at the [code for it](https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/modules/borg_buffs/code/robot.dm#L114). The sprites are [here](https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/modules/borg_buffs/icons/items_cyborg.dmi).

Gonna have a gander at that after I get home from work tomorrow, then ask the maintainers what they think of the sprites. For now putting it on draft so I can also try to have another crack at making an item that lets the borg open the cooking part of the crafting menu

Wallemations commented:
How does the rolling table dock work? Does it print a table or is it just for picking up and putting down a table?

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
Knifepin, service apparatus

Added the suggested knife/rolling pin power tool, but made some alterations such as it not being faster, and still keeping the whole 'no stabby' part, while also making the rolling pin sound blunt

Dawnseer commented:
Alright, went with the suggested knife/rolling pin tool, made some alterations, and also added the service apparatus, a tool for picking up food and oven trays. Still can't get the service borg to be able to craft. So I'll leave that to some more experienced coder. Undtil then, service borgs will have to settle for simply assisting in the kitchen.

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
service apparatus sprite, oven interaction

Gave the service apparatus it's own sprite, allowed borgs to open and close the oven. Still can't pick trays up from the oven. But you can place food in it, and cook it, and open the oven when it's done. Just not retrieve it yourself.

Dawnseer committed:
Merge branch 'ServiceBorgUpgrades' of https://github.com/Dawnseer/tgstation into ServiceBorgUpgrades

Dawnseer committed:
silicon knife to kitchen toolset

Dawnseer committed:
LINT

Dawnseer committed:
Added papercups to the RSF

... Sillycup? Really?

Dawnseer committed:
Whoops

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
Update hypo.dm

Added vinegar to the shaker, exchanged corn syrup for corn oil and corn starch in the condiment synthesizer

Dawnseer committed:
LINT

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer commented:
God damnit

Dawnseer committed:
Update kitchen.dmi

Got the brand spankin new grinder sprites to avoid the merge conflict

Overuseddart commented:
Can there be a portable grinder?

Dawnseer commented:
> Can there be a portable grinder?

Could probably get made. I was working on a portable oven that I decided to scrap after I couldn't figure out how to let the service apparatus pick up trays from the oven.

Dawnseer committed:
Update hypo.dm

C'mon check out already

Dawnseer committed:
Merge remote-tracking branch 'upstream/master' into ServiceBorgUpgrades

dragomagol commented on code/game/objects/items/kitchen.dm:
```diff
@@ -102,6 +102,45 @@
 	AddElement(/datum/element/easily_fragmented, PLASTIC_BREAK_PROBABILITY)
 
+/obj/item/knife/kitchen/silicon
+	name= "Kitchen Toolset"
```
```suggestion
	name = "Kitchen Toolset"
```

san7890 replied:
```suggestion
	name = "Kitchen Toolset"
```

nit

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+	model_type = list(/obj/item/robot_model/service)
+	model_flags = BORG_MODEL_SERVICE
+
+/obj/item/borg/upgrade/condimentsynthesizer/action(mob/living/silicon/robot/R, user = usr)
```
Should have a better variable name than `R` (even though everywhere else uses `R`, we still want to move away from it)

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+/obj/item/borg/upgrade/condimentsynthesizer/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/reagent_containers/borghypo/condimentsynthesizer/CS = locate() in R.model.modules
```
Same here, should have a better name (even something as simple as `synthesizer`)

san7890 replied:
don't do abbreviation vars in new code, we have to inherit `action` having the arg `R` unfortunately, but we shouldn't repeat the pattern in modern code. Change it to something like

```suggestion
		var/obj/item/reagent_containers/borghypo/condimentsynthesizer/synth = locate() in R.model.modules
```

or whatever you want really. abbreviations are just wack and this is no-where near being hot-enough to necessitate having abbreviations for var names.

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+		if (CS)
+			R.model.remove_module(CS, TRUE)
+			
+/obj/item/borg/upgrade/siliconknife
```
This shouldn't be in the silicon storage dm file (same with other added non-storage borg items below).

Dawnseer replied:
As far as I can see, only the service apparatus is in the silicon storage file?

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+			
+/obj/item/borg/upgrade/siliconknife
+	name = "Service Cyborg Kitchen Toolset"
+	desc = "An upgrade to the service model cyborg, to help process foods"
```
```suggestion
	desc = "An upgrade to the service model cyborg, to help process foods."
```

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+			
+/obj/item/borg/upgrade/siliconknife/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/knife/kitchen/silicon/SK = locate() in R.model.modules
```
Needs better variable name

san7890 replied:
ditto early return and abbrevation-var

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+			
+/obj/item/borg/upgrade/serviceapparatus
+	name = "Service Cyborg Service Apparatus"
+	desc = "An upgrade to the service model cyborg, to help handle foods"
```
```suggestion
	desc = "An upgrade to the service model cyborg, to help handle foods."
```

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+	if(.)
+		var/obj/item/knife/kitchen/silicon/SK = locate() in R.model.modules
+		if(SK)
+			to_chat(user, span_warning("This unit has no room for another Kitchen toolset!"))
```
```suggestion
			to_chat(user, span_warning("This unit has no room for another [src]!"))
```
I would test the above though, my spidey senses are telling me that because the name starts with a capital letter it might treat it grammatically as a proper noun.

dragomagol commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+			
+/obj/item/borg/upgrade/rollingtable
+	name = "Service Cyborg Rolling Table Dock"
+	desc = "An upgrade to the service model cyborg, to help provide mobile service"
```
```suggestion
	desc = "An upgrade to the service model cyborg, to help provide mobile service."
```

dragomagol commented on code/game/objects/items/rollertable_dock.dm:
```diff
@@ -0,0 +1,57 @@
+			RT.loaded = src
+			forceMove(RT)
+			user.visible_message(span_notice("[user] collects [src]."), span_notice("You collect [src]."))
+		return 1
```
```suggestion
		return TRUE
```

san7890 replied:
is this a boolean? return `TRUE` instead.

dragomagol reviewed (changes requested):
Mostly comments on variable names; you should be avoiding acronyms and single-letter variables. I would also suggest making your new object names snake_case, i.e. `condiment_synthesizer` and `service_apparatus`. 

I'm also inclined to say you should be using balloon alerts for messages such as 
```to_chat(user, span_warning("This unit has no room for another rolling table dock!"))```
becoming
```borg.balloon_alert_to_viewers("already installed!")```

san7890 commented on code/game/objects/items/kitchen.dm:
```diff
@@ -102,6 +102,45 @@
+
+/obj/item/knife/kitchen/silicon/examine()
+	. = ..()
+	. += " It's fitted with a [tool_behaviour == TOOL_KNIFE ? "knife" : "rolling pin"] head."
```
Okay, I know why you did it like this, but it should really be like this because of how `tool_behaviour` flags compile out to strings.

```suggestion
	. += " It's fitted with a [tool_behaviour] head."
```

Then, you should go to the file where we have `#define TOOL_ROLLINGPIN "rollingpin"` and swap it to `#define TOOL_ROLLINGPIN "rolling pin"`. This shouldn't have any adverse effects on the code pretty much anywhere (by design), and the benefit is clear.

dragomagol commented:
Oh and also please show new TGUI. 
And list your additions, I see you berry juice (about dang time) and cherry jelly.

san7890 commented on code/game/objects/items/kitchen.dm:
```diff
@@ -102,6 +102,45 @@
 
+/obj/item/knife/kitchen/silicon
+	name= "Kitchen Toolset"
+	icon_state = "SiliKnife"	
```
nit, but `icon_state` stuff should really be `snake_case`, so this would be `sili_knife` instead (it's developer facing only, so it's fine your flavor isn't lost). just change the name in the DMI and change it here as well

san7890 commented on code/game/objects/items/kitchen.dm:
```diff
@@ -102,6 +102,45 @@
+	if(tool_behaviour != TOOL_ROLLINGPIN)
+		tool_behaviour = TOOL_ROLLINGPIN
+		to_chat(user, span_notice("You attach the rolling pin bit to the [src]."))
+		icon_state = "SiliRollingpin"
```
especially ditto here with the `snake_case` stuff, this is very cluttered. `sili_rolling_pin` is a lot better.

san7890 commented on code/game/objects/items/rcd/RSF.dm:
```diff
@@ -32,10 +32,16 @@ RSF
 	w_class = WEIGHT_CLASS_NORMAL
 	///An associated list of atoms and charge costs. This can contain a separate list, as long as it's associated item is an object
 	var/list/cost_by_item = list(/obj/item/reagent_containers/cup/glass/drinkingglass = 20,
+								/obj/item/reagent_containers/cup/glass/sillycup = 10,
 								/obj/item/paper = 10,
 								/obj/item/storage/dice = 200,
 								/obj/item/pen = 50,
 								/obj/item/clothing/mask/cigarette = 10,
+								/obj/item/plate = 70,
+								/obj/item/reagent_containers/cup/bowl = 70,
+								/obj/item/kitchen/fork/plastic = 30,
+								/obj/item/knife/plastic = 30,
+								/obj/item/kitchen/spoon/plastic = 30,
 								)
```
this is a really bad list multiline and it's not your fault, but let's fix it.

```suggestion
	var/list/cost_by_item = list(
		/obj/item/reagent_containers/cup/glass/drinkingglass = 20,
		/obj/item/reagent_containers/cup/glass/sillycup = 10,
		/obj/item/paper = 10,
		/obj/item/storage/dice = 200,
		/obj/item/pen = 50,
		/obj/item/clothing/mask/cigarette = 10,
		/obj/item/plate = 70,
		/obj/item/reagent_containers/cup/bowl = 70,
		/obj/item/kitchen/fork/plastic = 30,
		/obj/item/knife/plastic = 30,
		/obj/item/kitchen/spoon/plastic = 30,
	)
```

it would also be great if you could alphabetize this (via a vscode extension) or arrange it by weights or whatever.

Dawnseer replied:
Honestly the way it's arranged right now is by use, like the glass and paper cup together, the paper, dice and pen together, cigarette... in there where it usually was, the plate and bowl together, and the plastic cutlery together. 
If you'd still prefer it being alphabetical or by weights or somesuch, do dell

san7890 replied:
does it really show up in this exact order in some player-facing UI? If so, put a comment delineating that and i'm happy because otherwise someone in the not-so-distant future will see this and go "silly coder didn't alphabetize!" and screw it all up.

should defo still be multi-lined though.

Dawnseer replied:
Yeah it shows up in this order in the playerfacing menu when using the RSF,  the same with the borgshaker and condiment synthesizer, the things written first shows first, the next comes after, etc etc.

Dawnseer replied:
So anyway I left a comment there specifying why it isn't alphabetical and did the multi-line thing. At least I think that's what you meant by multi line.

san7890 commented on code/game/objects/items/robot/items/hypo.dm:
```diff
@@ -52,23 +52,28 @@
 		/datum/reagent/consumable/ethanol/tequila, /datum/reagent/consumable/ethanol/triple_sec, /datum/reagent/consumable/ethanol/vermouth,\
 		/datum/reagent/consumable/ethanol/vodka, /datum/reagent/consumable/ethanol/whiskey, /datum/reagent/consumable/ethanol/wine\
 	)
+#define EXPANDED_SERVICE_REAGENTS list (/datum/reagent/consumable/flour, /datum/reagent/consumable/rice, /datum/reagent/consumable/salt,\
+		/datum/reagent/consumable/blackpepper, /datum/reagent/consumable/sugar, /datum/reagent/consumable/coco, /datum/reagent/consumable/vanilla,\
+		/datum/reagent/consumable/cornmeal, /datum/reagent/consumable/corn_starch, /datum/reagent/consumable/cornoil, /datum/reagent/consumable/eggwhite,\
+		/datum/reagent/consumable/eggyolk\
+	)
```
```suggestion
#define EXPANDED_SERVICE_REAGENTS list(\
	/datum/reagent/consumable/flour,\
	/datum/reagent/consumable/rice,\
	/datum/reagent/consumable/salt,\
	/datum/reagent/consumable/blackpepper,\
	/datum/reagent/consumable/sugar,\
	/datum/reagent/consumable/coco,\
	/datum/reagent/consumable/vanilla,\
	/datum/reagent/consumable/cornmeal,\
	/datum/reagent/consumable/corn_starch,\
	/datum/reagent/consumable/cornoil,\
	/datum/reagent/consumable/eggwhite,\
	/datum/reagent/consumable/eggyolk,\
)
```

ditto on alphabetization. if you could also patch `BASE_SERVICE_REAGENTS` in a similar manner since you touched that in this PR (we can leave the untouched ones for another day) that would be splendid!

Dawnseer replied:
Base service reagants are already alphabetized, the stuff I added was added in alphabetical manner after I noticed it already was as such, before I was beginning to make it more like 'all the juices together in one section, all the sodas in the other'

san7890 commented on code/game/objects/items/robot/items/hypo.dm:
```diff
@@ -370,6 +374,69 @@
+		balloon_alert(user, "no reagent selected!")
+		return .
+	. |= AFTERATTACK_PROCESSED_ITEM
+	if(target.is_refillable())
```
can we turn this into an early return like `if(!target.is_refillable())` then `return .`? would save a bit of ugly indentation.

san7890 commented on code/game/objects/items/robot/items/storage.dm:
```diff
@@ -300,3 +300,34 @@
+	name = "Service apparatus"
+	desc = "A special apparatus for carrying food and oven trays"
+	icon_state = "borg_service_apparatus"
+	storable = list(/obj/item/plate/oven_tray,
+				/obj/item/food)
```
```suggestion
	storable = list(
		/obj/item/plate/oven_tray,
		/obj/item/food,
	)
```

ditto alphabetize

san7890 commented on code/game/objects/items/robot/items/storage.dm:
```diff
@@ -300,3 +300,34 @@
+	. = ..()
+	if(stored)
+		. += "The apparatus currently has [stored] secured."
+	. += span_notice(" <i>Alt-click</i> will drop the currently secured item.")
```
```suggestion
	. += span_notice("<i>Alt-click</i> will drop the currently secured item.")
```

`examine()` should automatically new line this, right?

san7890 commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+
+/obj/item/borg/upgrade/condimentsynthesizer/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
```
```suggestion
	if(!.)
		return FALSE
		
	// move all your code to this indentation
```

san7890 commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+
+/obj/item/borg/upgrade/condimentsynthesizer/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
```
ditto early return

san7890 commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+/obj/item/borg/upgrade/condimentsynthesizer/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/reagent_containers/borghypo/condimentsynthesizer/CS = locate() in R.model.modules
```
ditto on the abbreviation-var

san7890 commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+
+/obj/item/borg/upgrade/siliconknife/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/knife/kitchen/silicon/SK = locate() in R.model.modules
```
ditto early return and abbreviation var

san7890 commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+
+/obj/item/borg/upgrade/serviceapparatus/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/borg/apparatus/service/RP = locate() in R.model.modules
```
ditto early return and abbreviation var

san7890 commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+			
+/obj/item/borg/upgrade/rollingtable/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/rollingtable_dock/RT = locate() in R.model.modules
```
ditto early return and abbreviation var

san7890 commented on code/game/objects/items/robot/robot_upgrades.dm:
```diff
@@ -750,6 +750,111 @@
+
+/obj/item/borg/upgrade/rollingtable/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/rollingtable_dock/RT = locate() in R.model.modules
```
ditto early return and abbreviation var

san7890 commented on code/game/objects/items/rollertable_dock.dm:
```diff
@@ -0,0 +1,57 @@
+	loaded = new(src)
+
+/obj/structure/table/rolling/attackby(obj/item/WT, mob/user, params)
+	if(istype(WT, /obj/item/rollingtable_dock))
```
you can make this an early return too, like

```suggestion
	if(!istype(WT, /obj/item/rollingtable_dock))
		return
		
	// code goes on this line
	// hey, no need for that else case anymore!
```

san7890 commented on code/game/objects/items/rollertable_dock.dm:
```diff
@@ -0,0 +1,57 @@
+/obj/item/rollingtable_dock/afterattack(obj/target, mob/user , proximity)
+	. = ..()
+	var/turf/target_turf = get_turf(target)
+	if(!proximity)
+		return
+	if(target_turf.is_blocked_turf(TRUE))
+		return
```
```suggestion
	if(!proximity || target_turf.is_blocked_turf(TRUE))
		return
```

san7890 replied:
shortcircuiting is cool btw https://www.byond.com/docs/ref/#/operator/||

san7890 commented on code/game/objects/items/rollertable_dock.dm:
```diff
@@ -0,0 +1,57 @@
+		deploy_rollingtable(user, target)
+
+/obj/item/rollingtable_dock/proc/deploy_rollingtable(mob/user, atom/location)
+	var/obj/structure/table/rolling/RT = new /obj/structure/table/rolling(location)
```
ditto abbreviation-var

san7890 commented on code/modules/research/designs/mechfabricator_designs.dm:
```diff
@@ -1148,6 +1148,50 @@
+	build_type = MECHFAB
+	build_path = /obj/item/borg/upgrade/rollingtable
+	materials = list(/datum/material/iron = 20000, /datum/material/titanium = 750) //steeper price than a regular rolling table, with some added titanium to make up for the relative rarity of regular rolling tables
+	construction_time = 40
```
assuming these are in deciseconds, you should really patch this up to `4 SECONDS` instead... let me know if it's in something wacky like ticks and we can do something else.

ditto change this in all the other designs introduced in thsi PR.

Dawnseer replied:
Works just dandy

san7890 reviewed (changes requested):
23 reviews seems like a lot but it's a lot of minor things repeated a lot

san7890 commented:
lol i started around the same time tattle did whoops. both of our reviews are cool though

san7890 commented:
hit that "Ready for review" when you inevitably get through the wall of reviews and it's in a re-reviewable state

Dawnseer commented:
Lord have mercy

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
AAAAAAAAAAAAA

Dawnseer committed:
Merge branch 'ServiceBorgUpgrades' of https://github.com/Dawnseer/tgstation into ServiceBorgUpgrades

Dawnseer committed:
LINT

Dawnseer committed:
MORE LINT

Dawnseer commented:
Alright, took a couple of minutes, but got through the list. There is one thing I haven't changed yet and that is the RSF list, due to the listed reason in the comment at that part, so get back to me on that please if that's okay or if you'd really prefer it to be alphabetized

Dawnseer committed:
Update kitchen.dm

Removing the drill use sound, it's silly to hear a drill when you're using a knife.

dragomagol reviewed (approved):
Love it

Dawnseer committed:
Multi-line?

Dawnseer committed:
comment

Just explaining above the code why it's not alphabetical, but rather sorted by category due to how it shows in player facing ui

Dawnseer committed:
Last minute changes

Enabled the borgshaker and condiment synthesizer to be able to pour 1 unit at a time, and made the service apparatus able to also hold plates, bowls, and soup pots.

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
More last minute changes

Adjusted so the first page of the RSF is the cups, bowl, plate and cutlery, while the second page is the misc stuff. Also added seaweed sheets to it.

Dawnseer committed:
Merge branch 'ServiceBorgUpgrades' of https://github.com/Dawnseer/tgstation into ServiceBorgUpgrades

Dawnseer committed:
Even more last minute changes.

How many minutes are you gonna give me? Anyway, adds blood and carpotoxin to the emagged borgshaker.

san7890 commented on code/game/objects/items/rcd/RSF.dm:
```diff
@@ -31,12 +31,22 @@ RSF
-								/obj/item/clothing/mask/cigarette = 10,
-								)
+	///The RSF item list below shows in the player facing ui in this order, this is why it isn't in alphabetical order, but rather sorted by category
+	var/list/cost_by_item = list(\
+		/obj/item/reagent_containers/cup/glass/drinkingglass = 20,\
+		/obj/item/reagent_containers/cup/glass/sillycup = 10,\
+		/obj/item/plate = 70,\
+		/obj/item/reagent_containers/cup/bowl = 70,\
+		/obj/item/kitchen/fork/plastic = 30,\
+		/obj/item/knife/plastic = 30,\
+		/obj/item/kitchen/spoon/plastic = 30,\
+		/obj/item/food/seaweedsheet = 30,\
+		/obj/item/storage/dice = 200,\
+		/obj/item/toy/cards/deck = 200,\
+		/obj/item/paper = 10,\
+		/obj/item/pen = 50,\
+		/obj/item/clothing/mask/cigarette = 10,\
+	)
```
```suggestion
	var/list/cost_by_item = list(
		/obj/item/reagent_containers/cup/glass/drinkingglass = 20,
		/obj/item/reagent_containers/cup/glass/sillycup = 10,
		/obj/item/plate = 70,
		/obj/item/reagent_containers/cup/bowl = 70,
		/obj/item/kitchen/fork/plastic = 30,
		/obj/item/knife/plastic = 30,
		/obj/item/kitchen/spoon/plastic = 30,
		/obj/item/food/seaweedsheet = 30,
		/obj/item/storage/dice = 200,
		/obj/item/toy/cards/deck = 200,
		/obj/item/paper = 10,
		/obj/item/pen = 50,
		/obj/item/clothing/mask/cigarette = 10,
	)
```

You only need to add the trailing `\` when it's a list tied to a `#define` macro, native lists such as these don't need that and can be multi-lined just fine.

san7890 commented on code/game/objects/items/robot/items/hypo.dm:
```diff
@@ -52,26 +52,42 @@
 		/datum/reagent/toxin/fakebeer,\
-		/datum/reagent/consumable/ethanol/fernet\
+		/datum/reagent/consumable/ethanol/fernet,\
 	)
```
```suggestion
)
```

san7890 replied:
i go into why we do it like this in the next review

san7890 commented on code/game/objects/items/robot/items/storage.dm:
```diff
@@ -300,3 +300,40 @@
+	name = "Service apparatus"
+	desc = "A special apparatus for carrying food, bowls, plates, oven trays, soup pots and paper."
+	icon_state = "borg_service_apparatus"
+	storable = list(
+				/obj/item/reagent_containers/cup/bowl,\
+				/obj/item/food,\
+				/obj/item/plate/oven_tray,\
+				/obj/item/paper,\
+				/obj/item/plate,\
+				/obj/item/reagent_containers/cup/soup_pot,\
+)
```
```suggestion
	storable = list(
		/obj/item/reagent_containers/cup/bowl,
		/obj/item/food,
		/obj/item/plate/oven_tray,
		/obj/item/paper,
		/obj/item/plate,
		/obj/item/reagent_containers/cup/soup_pot,
	)
```

each indented entry should only be one indent greater than the indent of the line the `list()` was started on. since line 308 has one indent, every entry of the list should have two indents. we also have the close parentheses at the same indent level of the line that the `list()` was started on.

(also ditto on the whole backslash thing)

san7890 reviewed (changes requested):
do this and we're solid

Dawnseer committed:
Anything for you

Dawnseer committed:
Merge branch 'master' into ServiceBorgUpgrades

Dawnseer committed:
Just making it check again

Dawnseer commented:
> This is really cool! I played around with it, and noticed that borgs are unable to do things like remove meat slabs from the griddle even though they can put them on there. Same thing with putting stuff in the fryer - they can't be taken out.

Indeed, which is why i got a code bounty out for that since I couldn't solve it myself
